packages/dd-trace/src/llmobs/telemetry.js

```diff
@@ -53,7 +61,45 @@ function incrementLLMObsSpanFinishedCount (span, value = 1) {
   llmobsMetrics.count('span.finished', tags).inc(value)
 }
 
+function submitLLMObsRawSpanSize (event, rawEventSize) {
+  const spanKind = event.meta?.spanKind || ''
+  const integration = extractIntegrationFromTags(event.tags)
+  const error = event.status === 'error'
+  const autoinstrumented = integration != null
+
+  const tags = {
+    span_kind: spanKind,
+    autoinstrumented: Number(autoinstrumented),
+    error: error ? 1 : 0,
+    integration: integration || 'N/A'
+  }
+
+  llmobsMetrics.distribution('span.raw_size', tags).track(rawEventSize)
+}
+
+function submitLLMObsSpanSize (event, eventSize) {
+  const spanKind = event.meta?.spanKind || ''
+  const integration = extractIntegrationFromTags(event.tags)
+  const error = event.status === 'error'
+  const autoinstrumented = integration != null
+  const truncated = (
+    Array.isArray(event.collection_errors) && event.collection_errors.includes(DROPPED_IO_COLLECTION_ERROR)
+  )
+
+  const tags = {
+    span_kind: spanKind,
+    autoinstrumented: Number(autoinstrumented),
+    error: error ? 1 : 0,
+    integration: integration || 'N/A',
+    truncated: Number(truncated)
+  }
+
+  llmobsMetrics.distribution('span.size', tags).track(eventSize)
```

Comment:
could be worth extracting some common logic out from these functions, i see we have smth similar in python where we have a shared function for grabbing the tags (which could be helpful if you add more telemetry for span events with base tags). we could also possibly pass in a `truncated` parameter and just have one function that sets the truncated tag and determines which telemetry event name to use. purely stylistic tbh - your call!

Reply:
Ahh great suggestion, will simplify this!